src/Storages/Distributed/DistributedAsyncInsertSource.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
```

Comment:
Do we have to put this file under `Distributed` dir?

Reply:
Well, this is not strictly required, but why not?
There is also `src/Storages/Distributed/DistributedSettings.h`